### PR TITLE
Feature/survey config

### PR DESCRIPTION
**Pull request by prashantesmagico**

### CODE REVIEW

File: prisma/migrations/20231010061334_survey_config/migration.sql
Comment (Ajay-Maury) [inline prisma/migrations/20231010061334_survey_config/migration.sql:1]: Remove migration files for now

File: prisma/migrations/migration_lock.toml
Comment (Ajay-Maury) [inline prisma/migrations/migration_lock.toml:1]: Remove migration files for now

File: src/main.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ async function bootstrap() {
   app.useGlobalPipes(new ValidationPipe(validationOptions));
 
   // Configure global serialization using class-transformer
-  app.useGlobalInterceptors(new ClassSerializerInterceptor(app.get(Reflector)));
+  // app.useGlobalInterceptors(new ClassSerializerInterceptor(app.get(Reflector)));
```
Comment (Ajay-Maury) [inline src/main.ts:42]: Uncomment this code and add proper return type for APIs

File: src/survey-config/dto/create-survey-config.dto.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  //Optional endTime filter validate that its valid date-time value.
+  @IsOptional()
+  @IsNotEmpty()
+  @IsDate()
+  endTime?: Date;
+
+  // Optional limit for pagination, validate that it's an integer.
+  @IsOptional()
+  @IsInt()
+  limit?: number;
```
Comment (Ajay-Maury) [inline src/survey-config/dto/create-survey-config.dto.ts]: Add some default value for limit and offset

File: src/survey-config/entities/survey-config.entity.ts
Comment (Ajay-Maury) [inline src/survey-config/entities/survey-config.entity.ts:1]: Remove this entity file as it in not used anywhere

File: src/survey-config/survey-config.controller.ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+  // The survey config controller class, injecting the survey config service.
+  constructor(private readonly surveyConfigService: SurveyConfigService) {}
+
+  @Post()
+  @ApiOperation({ summary: "Create a survey config" }) // Describes the api operation for Swagger.
+  @ApiResponse({ status: HttpStatus.CREATED, type: ResponseSurveyConfigDto }) // Describes the response for Swagger.
+  async createSurveyConfig(
+    @Res() res,
+    @Body() createSurveyConfigDto: CreateSurveyConfigDto
+  ) {
```
Comment (Ajay-Maury) [inline src/survey-config/survey-config.controller.ts]: Add return type for all APIs

File: src/survey-config/survey-config.controller.ts [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+      // Return a success response with the created request data
+      return res.status(HttpStatus.CREATED).json({
+        message: "Survey config created successfully",
+        data: createdSurveyConfig,
+      });
+    } catch (error) {
+      this.logger.error(`Failed to create new survey config.`, error);
+
+      // Return an error response
+      return res.status(HttpStatus.CREATED).json({
```
Comment (Ajay-Maury) [inline src/survey-config/survey-config.controller.ts]: In error status can't be created

File: src/survey-config/survey-config.controller.ts [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    }
+  }
+
+  // Update the survey config
+  @Patch("/update/:id")
+  @ApiOperation({ summary: "Update the survey config by ID" }) // Api operation for swagger
+  @ApiResponse({ status: HttpStatus.OK, type: ResponseSurveyConfigDto }) // Api operation for swagger
+  async updateSurveyConfigById(
+    @Res() res,
+    @Param("id") id: number,
```
Comment (Ajay-Maury) [inline src/survey-config/survey-config.controller.ts]: Add parseInt in params and remove + from all Ids

File: src/survey-config/survey-config.controller.ts [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+      return res.status(HttpStatus.CREATED).json({
+        message: "Survey config created successfully",
+        data: createdSurveyConfig,
+      });
+    } catch (error) {
+      this.logger.error(`Failed to create new survey config.`, error);
+
+      // Return an error response
+      return res.status(HttpStatus.CREATED).json({
+        message: `Failed to create new survey config.`,
```
Comment (Ajay-Maury) [inline src/survey-config/survey-config.controller.ts]: Add prisma error message and if not then custom error message in all APIs

**Review by Ajay-Maury — CHANGES_REQUESTED** [review]

File: prisma/migrations/20231010061334_survey_config/migration.sql
Comment (prashantesmagico) [reply prisma/migrations/20231010061334_survey_config/migration.sql:1]: Will remove in the next commit

File: src/survey-config/dto/create-survey-config.dto.ts (hunk 2 above)
Comment (prashantesmagico) [reply src/survey-config/dto/create-survey-config.dto.ts]: Done

File: src/survey-config/entities/survey-config.entity.ts
Comment (prashantesmagico) [reply src/survey-config/entities/survey-config.entity.ts:1]: removed

File: src/survey-config/survey-config.controller.ts (hunk 4 above)
Comment (prashantesmagico) [reply src/survey-config/survey-config.controller.ts]: Made the changes

File: src/survey-config/survey-config.controller.ts (hunk 5 above)
Comment (prashantesmagico) [reply src/survey-config/survey-config.controller.ts]: Done

File: src/main.ts [hunk 7]
```diff
@@ -56,6 +56,8 @@ async function bootstrap() {
     .setDescription(SWAGGER_CONSTANTS.DESCRIPTION)
     .setVersion(SWAGGER_CONSTANTS.VERSION)
     .addTag(SWAGGER_TAGS.DEFAULT) // Add a tag for API grouping
+    .addTag(SWAGGER_TAGS.SURVEYCONFIG)
+    .addTag(SWAGGER_TAGS.SURVEYPARAMETER)
```
Comment (VamshiBatta07) [inline src/main.ts]: Please remove this piece of code. Not Required.

File: src/survey-parameter/dto/create-survey-parameter.dto.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  // Optional limit for pagination, validate that it's an integer.
+  @IsOptional()
+  @IsInt()
+  limit?: number = 10;
+
+  // Optional offset for pagination, validate that it's an integer.
+  @IsOptional()
+  @IsInt()
+  offset?: number = 0;
+}
```
Comment (VamshiBatta07) [inline src/survey-parameter/dto/create-survey-parameter.dto.ts]: Please remove this.

File: src/utils/constants.ts [hunk 9]
```diff
@@ -13,4 +13,6 @@ export const SWAGGER_CONSTANTS = {
 
 export enum SWAGGER_TAGS {
   DEFAULT = "default",
+  SURVEYCONFIG = "survey-config",
+  SURVEYPARAMETER = "survey-parameter"
```
Comment (VamshiBatta07) [inline src/utils/constants.ts]: Please remove this.

**Review by VamshiBatta07 — CHANGES_REQUESTED** [review]

File: src/main.ts (hunk 7 above)
Comment (prashantesmagico) [reply src/main.ts]: removed

File: src/utils/constants.ts (hunk 9 above)
Comment (prashantesmagico) [reply src/utils/constants.ts]: removed

File: src/survey-parameter/dto/create-survey-parameter.dto.ts (hunk 8 above)
Comment (prashantesmagico) [reply src/survey-parameter/dto/create-survey-parameter.dto.ts]: removed

File: src/survey-config/survey-config.controller.ts (hunk 3 above)
Comment (prashantesmagico) [reply src/survey-config/survey-config.controller.ts]: Added return type for all APIs

**Comment by Ajay-Maury** [comment]
Raised another PR for this:- https://github.com/COMPASS-DPG/wpcas-service/pull/18

**Comment by Ajay-Maury** [comment]
Raised another PR for this:- https://github.com/COMPASS-DPG/wpcas-service/pull/18

**Pull request closed without merging**